tsdb/tsm1/engine_delete_prefix.go

```diff
@@ -245,34 +246,53 @@ func (e *Engine) DeletePrefixRange(rootCtx context.Context, name []byte, min, ma
 			// and remove from the series file.
 			span, _ = tracing.StartSpanFromContextWithOperationName(rootCtx, "SFile Delete Series IDs")
 			span.LogKV("measurement_name", fmt.Sprintf("%x", name), "series_id_set_size", set.Cardinality())
-			set.ForEachNoLock(func(id tsdb.SeriesID) {
-				if err = e.sfile.DeleteSeriesID(id); err != nil {
-					return
-				}
-			})
+			var ids []tsdb.SeriesID
+			set.ForEachNoLock(func(id tsdb.SeriesID) { ids = append(ids, id) })
+			if err = e.sfile.DeleteSeriesIDs(ids); err != nil {
+				return err
+			}
 			span.Finish()
 			return err
 		}
 
 		// This is the slow path, when not dropping the entire bucket (measurement)
 		span, _ = tracing.StartSpanFromContextWithOperationName(rootCtx, "TSI/SFile Delete keys")
 		span.LogKV("measurement_name", fmt.Sprintf("%x", name), "keys_to_delete", len(possiblyDead.keys))
+
+		// Convert key map to a slice.
+		possiblyDeadKeysSlice := make([][]byte, 0, len(possiblyDead.keys))
 		for key := range possiblyDead.keys {
-			// TODO(jeff): ugh reduce copies here
-			keyb := []byte(key)
-			keyb, _ = SeriesAndFieldFromCompositeKey(keyb)
+			possiblyDeadKeysSlice = append(possiblyDeadKeysSlice, []byte(key))
+		}
 
-			name, tags := models.ParseKeyBytes(keyb)
-			sid := e.sfile.SeriesID(name, tags, buf)
-			if sid.IsZero() {
-				continue
+		const batchSize = 1000
```

Comment:
This is roughly 125 keys per TSI/sfile partition per batch. Any performance difference between 1,000 and 10,000?

Reply:
I'm not sure. The performance increase is already several orders of magnitude faster and the trade off would be using more memory. I'm not sure it's worth increasing without an example of slow deletes (after this change).